### PR TITLE
DEV-4188: Remove spin behavior from amount field

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Changes the amount field from `type="number"` to `type="text" inputmode="numeric" pattern="\d+"`.

#### Why are we doing this? How does it help us?

Prevents inadvertent changes to the amount field.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Updated existing.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-4188

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/src/components/donationPage/pageContent/DAmount.test.tsx [hunk 1]
```diff
@@ -300,29 +299,29 @@ describe('DAmount', () => {
       };
 
       tree(propsWithOtherAmount, pageContext);
-      fireEvent.change(screen.getByRole('spinbutton'), { target: { value: '1' } });
-      expect(screen.getByRole('spinbutton')).toHaveValue(1);
+      fireEvent.change(screen.getByRole('textbox'), { target: { value: '1' } });
+      expect(screen.getByRole('textbox')).toHaveValue('1');
       expect(screen.queryByTestId('amount-1-selected')).not.toBeInTheDocument();
     });
```
Comment (GuiMend) [inline spa/src/components/donationPage/pageContent/DAmount.test.tsx:306]: **suggestion (test improvement non-blocking):** Do you want to add tests to assert the behavior we are after?

Something like:
-> focus input
-> userEvent.type arrow-down
-> value didn't change

Ref: https://stackoverflow.com/questions/64492491/how-to-test-which-input-has-the-focus-when-tab-button-is-pressed

**Review by GuiMend — APPROVED** [review]
Code looks good. Have one suggestion for increasing tests

File: spa/src/components/donationPage/pageContent/DAmount.test.tsx (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:306]: I like this idea but I don't think jsdom supports this interaction. Here's what I tried to prove it out:

```tsx
import userEvent from "@testing-library/user-event";
import { render, screen } from "test-utils";

describe('Number input', () => {
  it('changes value when the up arrow is pressed?', async () => {
    render(<input type="number" />);
    userEvent.type(screen.getByRole('spinbutton'), '123');
    expect(screen.getByRole('spinbutton')).toHaveValue(123);
    expect(screen.getByRole('spinbutton')).toHaveFocus();
    userEvent.type(screen.getByRole('spinbutton'), '{arrowdown}');
    expect(screen.getByRole('spinbutton')).toHaveValue(122);
  });
});
```

Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:306]: How about in cypress @nrh-cklimas ? Perhaps it's doable there. I tried doing something similar to your example, but also didn't manage to make it work

Comment (nrh-cklimas) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:306]: Good idea--I was able to replicate the arrow key behavior in Cypress but not mouse wheel, but they're similar interactions so I'd think if one fails, the other would too. 611a346feb6287588c069ac809c390c548298687

**Review by GuiMend — APPROVED** [review]

**Comment by nrh-cklimas** [comment]
Merging with failed check because it's flagging a Python dep unrelated to this PR.

**Pull request merged**